src/metrics_probe_queue_lengths.erl

```diff
@@ -0,0 +1,122 @@
+%%==============================================================================
+%% Copyright 2019 Erlang Solutions Ltd.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%% http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%==============================================================================
+-module(metrics_probe_queue_lengths).
+-behaviour(exometer_probe).
+
+-include("mongoose_logger.hrl").
+
+%% exometer_entry callbacks
+%% exometer_probe callbacks
+-export(
+   [
+    behaviour/0,
+    probe_init/3,
+    probe_terminate/1,
+    probe_get_value/2,
+    probe_get_datapoints/1,
+    probe_update/2,
+    probe_reset/1,
+    probe_sample/1,
+    probe_setopts/3,
+    probe_handle_msg/2,
+    probe_code_change/3
+   ]).
+
+-define(DATAPOINTS, [regular, fsm, total]).
+
+-type datapoint() :: atom().
+-record(state, {
+    datapoints = [] :: [datapoint()],
+    data = #{} :: #{datapoint() => integer()},
+    ref :: reference() | undefined
+}).
+
+-spec behaviour() -> exometer:behaviour().
+behaviour() ->
+    probe.
+
+probe_init(_Name, _Type, _Opts) ->
+    {ok, #state{datapoints = ?DATAPOINTS, data = #{}}}.
+
+probe_terminate(_) -> ok.
+
+probe_get_value(DPs, #state{data = Data} = S) ->
+    {ok, probe_get_value_(Data, DPs), S}.
+
+probe_get_value_(Data, DPs) ->
```

Comment:
It's not really consistent with the majority of MIM code (`_` suffix). Also extracting this into a helper isn't that necessary.